generic/Storage.c

```diff
@@ -39,6 +39,27 @@ static int torch_Storage_(new)(lua_State *L)
       lua_pop(L, 1);
     }
   }
+  else if(lua_type(L, index) == LUA_TUSERDATA)
+  {
+    THStorage *src = luaT_checkudata(L, index, torch_Storage);
+    if (src->flag & TH_STORAGE_VIEW) {
+      // don't allow chained views
+      src = src->view;
```

Comment:
But that doesn't seem very intuitive to me. Creating a view on A is different from creating a view on B, because A may have an offset into B, which is ignored in this case, isn't it?